apps/api/src/modules/archived-notifications/archived-notifications.service.ts

```diff
@@ -0,0 +1,105 @@
+      archivedNotification.notificationId = notification.id;
+      archivedNotification.providerId = notification.providerId;
+      archivedNotification.result = notification.result;
+      archivedNotification.retryCount = notification.retryCount;
+      archivedNotification.updatedBy = notification.updatedBy;
+      archivedNotification.updatedOn = notification.updatedOn;
+      archivedNotification.status = notification.status;
+
+      this.logger.debug(
+        `Preparing ArchivedNotification entry using NotificationID: ${archivedNotification.notificationId}, DeliveryStatus: ${archivedNotification.deliveryStatus}, ApplicationID: ${archivedNotification.applicationId}, ProviderID: ${archivedNotification.providerId}`,
+      );
+
+      return archivedNotification;
+    });
+  }
+
+  async moveNotificationsToArchive(): Promise<void> {
+    const archiveLimit = this.configService.get<number>('ARCHIVE_LIMIT', 1000);
+
+    try {
+      // Step 1: Retrieve the notifications to archive
+      this.logger.log(`Retrieve the top ${archiveLimit} notifications to archive`);
+      const notificationsToArchive =
+        await this.notificationsService.findNotificationsToArchive(archiveLimit);
+
+      if (notificationsToArchive.length === 0) {
+        this.logger.log('No notifications to archive at this time.');
+        return;
+      }
+
+      this.logger.log('Creating queryRunner and starting transaction');
+      const queryRunner: QueryRunner = this.dataSource.createQueryRunner();
+      await queryRunner.connect();
+      await queryRunner.startTransaction();
+
+      try {
+        // Step 2: Convert notifications to archived notifications
+        const archivedNotificationsArray =
+          this.convertToArchivedNotifications(notificationsToArchive);
+
+        // Step 3: Insert notifications into the archive table
+        this.logger.log(`Inserting archived notifications into the archive table`);
+        await queryRunner.manager.save(ArchivedNotification, archivedNotificationsArray);
+
+        // Step 4: Delete notifications from the main table
+        this.logger.log(`Deleting notifications from the main table`);
+        const idsToDelete = notificationsToArchive.map((notification) => notification.id);
+        this.logger.log(`Notification IDs to delete: ${idsToDelete}`);
+        await queryRunner.manager.delete(Notification, idsToDelete);
+
+        await queryRunner.commitTransaction();
+        this.logger.log('Transaction successful');
+      } catch (error) {
+        this.logger.error('Error while archiving notifications. Rolling back transaction');
+        await queryRunner.rollbackTransaction();
+        throw error;
+      } finally {
+        await queryRunner.release();
+      }
+    } catch (error) {
+      this.logger.error(`Failed to archive notifications: ${error.message}`);
+      throw error;
+    }
+  }
+
+  @Cron(CronExpression.EVERY_HOUR)
+  async archiveCompletedNotificationsCron(): Promise<void> {
```

Comment:
@kshitij-k-osmosys 
Move this code to scheduler service.

Also since there is issue with Scheduler we were facing, we decided to use the scheduler script.
So you would need similar approach for triggering it.

Reply:
Not sure if we can allow users to set cron if it is triggering from scheduler script

Reply:
Update: tracking this in #360